### PR TITLE
Subs 996 data model refactor

### PR DESCRIPTION
**Pull request by neilgoodgame**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java [hunk 1]
```diff
@@ -47,114 +40,80 @@ public void setBaseSubmittable(Submittable baseSubmittable) throws IllegalAccess
...
-        }
-        return attributeValueList;
-    }
-
-    void deleteAttribute(Attribute attribute) {
-        getAttributes().remove(attribute);
-    }
-
     public void serialiseAttributes () throws IllegalAccessException {
+        final Set<String> strings = getAttributes().keySet();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java]: We are not using this variable later. Do we need it?

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java [hunk 2]
```diff
@@ -47,114 +40,80 @@ public void setBaseSubmittable(Submittable baseSubmittable) throws IllegalAccess
...
-            if (!enaFieldAttribute.fieldName().equals(ENAFieldAttribute.NO_FIELD)) {
-                fieldName = enaFieldAttribute.fieldName();
-            }
-             else {
-                fieldName = enaFieldAttribute.attributeName();
-            }
-            parseField(enaFieldAttribute, fieldName);
-        }
+    private void parseENAFieldAttributes (ENAValidation enaValidation) {
+        Map<String, String> attributefieldMap = new HashMap<>();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java:60]: We call `getAttributes()` method a number of times in this method and the `validate()` method, too. We could call it only ones here and use/pass that local variable later.

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java [hunk 3]
```diff
@@ -47,114 +40,80 @@ public void setBaseSubmittable(Submittable baseSubmittable) throws IllegalAccess
...
-    private void parseControlledAttributes (ENAValidation enaValidation) {
-        for (ENAControlledValueAttribute enaControlledValueAttribute : enaValidation.enaControlledValueAttributes()) {
-            String attributeName = enaControlledValueAttribute.attributeName();
-            List<String> allowedValueList = Arrays.asList(enaControlledValueAttribute.allowedValues());
-            final Optional<Attribute> optionalAttribute = getExistingAttribute(attributeName);
-            if (optionalAttribute.isPresent())
-                parseControlledValue(optionalAttribute.get(), allowedValueList);
+    private void parseControlledValue(String name, List<String> allowedValueList) {
+        if (getAttributes().containsKey(name)) {
+            for (Iterator<Attribute> it = getAttributes().get(name).iterator(); it.hasNext(); ) {
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java:102]: Can we use java8 here? Something like:
`getAttributes().get(name).forEach( attribute -> checkAllowedValues(name, allowedValueList, attribute));`

File: src/test/java/uk/ac/ebi/subs/ena/processor/ENAStudyProcessorTest.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public void processSubmission() throws Exception {
         final Team team = TestHelper.getTeam("test-team");
         final Study study = TestHelper.getStudy(alias, team,"study_abstract", "Whole Genome Sequencing");
         final LocalDate date = LocalDate.parse("2018-12-25");
-        study.setReleaseDate(date);
+        //study.setReleaseDate(date);
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/ena/processor/ENAStudyProcessorTest.java]: Could we remove these, please?

File: src/test/java/uk/ac/ebi/subs/xml/SerialisationTest.java [hunk 5]
```diff
@@ -180,11 +180,11 @@ protected void serialiseDeserialiseTest (String studyResource,
...
         final BaseSubmittable baseSubmittableFromResource = getBaseSubmittableFromResource(studyResource, baseSubmittableClass);
         final BaseSubmittable baseSubmittableForCompare = getBaseSubmittableFromResource(studyResource, baseSubmittableClass);
-        if (baseSubmittableClass.equals(Study.class)) {
-            ((Study)baseSubmittableForCompare).setReleaseDate(null);
+        if (baseSubmittableClass.equals(Project.class)) {
+            ((Project)baseSubmittableForCompare).setReleaseDate(null);
         }
 
-        Collections.sort(baseSubmittableForCompare.getAttributes());
+        sortAttributes(baseSubmittableForCompare);
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/xml/SerialisationTest.java]: Could we cleanup these, please?

**Review by ke4 — CHANGES_REQUESTED** [review]
There are just small changes

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java (hunk 1 above)
Comment (neilgoodgame) [reply src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java]: We are not using it.

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java (hunk 3 above)
Comment (neilgoodgame) [reply src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java:102]: I find it more readable without a stream

File: src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java (hunk 2 above)
Comment (neilgoodgame) [reply src/main/java/uk/ac/ebi/subs/data/submittable/AbstractENASubmittable.java:60]: Will leave this as it is.

**Review by ke4 — APPROVED** [review]

**Pull request merged**